File: frontend/src/contracts/enums.ts
// Using const objects to avoid erasableSyntaxOnly (enum) issues
export const UserRole = {
    PUBLIC_KIOSK: 'PUBLIC_KIOSK',
    TRAVELER: 'TRAVELER',
    HOST: 'HOST',
    MODERATOR: 'MODERATOR',
    ADMIN: 'ADMIN',
    SUPER_ADMIN: 'SUPER_ADMIN'
} as const;

export type UserRole = typeof UserRole[keyof typeof UserRole];

export const KYCStatus = {
    UNVERIFIED: 'UNVERIFIED',
    PENDING: 'PENDING',
    VERIFIED: 'VERIFIED',
    REJECTED: 'REJECTED'
} as const;

export type KYCStatus = typeof KYCStatus[keyof typeof KYCStatus];

File: backend/shared/contracts/enums.py
from django.db import models

class UserRole(models.TextChoices):
    PUBLIC_KIOSK = 'PUBLIC_KIOSK', 'Public Kiosk'
    TRAVELER = 'TRAVELER', 'Traveler'
    HOST = 'HOST', 'Host'
    MODERATOR = 'MODERATOR', 'Moderator'
    ADMIN = 'ADMIN', 'Admin'
    SUPER_ADMIN = 'SUPER_ADMIN', 'Super Admin'

class KYCStatus(models.TextChoices):
    UNVERIFIED = 'UNVERIFIED', 'Unverified'
    PENDING = 'PENDING', 'Pending'
    VERIFIED = 'VERIFIED', 'Verified'
    REJECTED = 'REJECTED', 'Rejected'
